app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -158,10 +166,27 @@ def set_pre_prov_vars
     @edit[:vm_sortcol] ||= "name"
     @edit[:prov_type] = "VM Provision"
     @edit[:hide_deprecated_templates] = true if request.parameters[:controller] == "vm_cloud"
+
     unless %w(image_miq_request_new miq_template_miq_request_new).include?(params[:pressed])
+      path_to_report = ManageIQ::UI::Classic::Engine.root.join("product", "views", "ProvisionTemplates.yaml").to_s
+      @view = MiqReport.new(YAML.load(File.open(path_to_report)))
+      @view.db = get_template_kls.to_s
+      report_scopes = %i(eligible_for_provisioning non_deprecated)
+      options = {
+        :model          => @view.db,
+        :gtl_type       => "table",
+        :named_scope    => report_scopes,
+        :path_to_report => path_to_report,
+      }
+      @custom_action = {
```

Comment:
why introduce a new instance variable for this?

Can't we just add it to options?

Reply:
Moved inside additional options